### DISCUSSION TITLE
V3 Repeater: Adding an extra actions near of the current actions

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9904] [Help]
same feature exist in

https://filamentphp.com/docs/3.x/actions/modals#adding-an-extra-modal-action-button-to-the-footer

Repeater have these action objects

```
->addAction()
->addBetweenAction()
->cloneAction()
->collapseAction()
->collapseAllAction()
->deleteAction()
->expandAction()
->expandAllAction()
->moveDownAction()
->moveUpAction()
->reorderAction()
```

but what if I want to add my own actions?

**Expected Result:**

```
use Filament\Actions\Action;

->extraActions([
  Action::make('undo')
    ->icon('bi-arrow-counterclockwise')
    ->iconButton(),

  Action::make('redo')
    ->icon('bi-arrow-clockwise')
    ->iconButton(),
]);
```

<img width="1242" alt="Screenshot 2023-11-27 at 13 14 10" src="https://github.com/filamentphp/filament/assets/121255432/a982a54d-475c-47d4-a1f4-f80575752c13">

**Resources:**

https://filamentphp.com/docs/3.x/forms/fields/repeater#customizing-the-repeater-action-objects

https://filamentphp.com/docs/3.x/actions/trigger-button

**Comment by ariefng** [comment]
You can use ->extraItemActions()

Real example is on Filament Demo https://github.com/filamentphp/demo/blob/main/app/Filament/Resources/Shop/OrderResource.php line 334

Result
![image](https://github.com/filamentphp/filament/assets/77451095/b247cf51-b8df-4c9d-9a01-17f885b56173)

Hope it helps.
